src/compute/flexbox.rs

```diff
@@ -974,13 +973,13 @@ fn calculate_children_base_lines(
 ) {
     /// Recursively calculates the baseline for children
     fn calc_baseline(db: &impl LayoutTree, node: Node, layout: &Layout) -> f32 {
-        let children = db.children(node);
-        if children.is_empty() {
+        if db.num_children(node) == 0 {
             layout.size.height
         } else {
-            let child = children[0];
-            let layout = db.layout(child);
-            calc_baseline(db, child, layout)
+            // Safe to unwrap because we already checked that the num_children > 0
+            let first_child = db.children(node).nth(0).unwrap();
+            let layout = db.layout(*first_child);
+            calc_baseline(db, *first_child, layout)
         }
```

Comment:
I think we can avoid the `unwrap` here by doing something like this:

```rs
if let Some(first_child) = db.children(node).next() {
    let layout = db.layout(*first_child);
    calc_baseline(db, *first_child, layout)
} else {
    layout.size.height
}
```

Reply:
I'm not sure if that's better or worse for performance though.

Reply:
Yep that's a good idea. I only didn't do this because I was trying to make minimal changes to the original source, but if people are happy with this then we can certainly replace the unwrap with an `if let` here.

Reply:
I'm happy to make that change in this PR.